addOns/spider/src/main/java/org/zaproxy/addon/spider/parser/SpiderGitParser.java

```diff
@@ -0,0 +1,372 @@
+                entryBytesRead += 4;
+                int indexEntryDev = dataBuffer.getInt();
+                entryBytesRead += 4;
+                int indexEntryInode = dataBuffer.getInt();
+                entryBytesRead += 4;
+                int indexEntryMode = dataBuffer.getInt();
+                entryBytesRead += 4;
+                int indexEntryUid = dataBuffer.getInt();
+                entryBytesRead += 4;
+                int indexEntryGid = dataBuffer.getInt();
+                entryBytesRead += 4;
+                int indexEntrySize = dataBuffer.getInt();
+                entryBytesRead += 4;
+                if (getLogger().isDebugEnabled())
+                    getLogger().debug("Entry " + entryIndex + " has size " + indexEntrySize);
+
+                // size is unspecified for the entry id, but it seems to be 40 bytes SHA-1 string
+                // stored as 20 bytes, network order
+                byte[] indexEntryIdBuffer = new byte[20];
+                dataBuffer.get(indexEntryIdBuffer, 0, 20);
+                entryBytesRead += 20;
+                String indexEntryId = new String(indexEntryIdBuffer);
+
+                short indexEntryFlags = dataBuffer.getShort();
+                entryBytesRead += 2;
+                if (getLogger().isDebugEnabled())
+                    getLogger().debug("Entry " + entryIndex + " has flags " + indexEntryFlags);
+
+                // mask off all but the least significant 12 bits of the index entry flags to get
+                // the length of the name in bytes
+                int indexEntryNameByteLength = indexEntryFlags & 4095;
+                if (getLogger().isDebugEnabled())
+                    getLogger()
+                            .debug(
+                                    "Entry "
+                                            + entryIndex
+                                            + " has a name of length "
+                                            + indexEntryNameByteLength);
+
+                // mask off all but the second most significant 12 bit of the index entry flags to
+                // get the extended flag for the entry
+                // int indexEntryExtendedFlag = indexEntryFlags & (int)16384;
+                int indexEntryExtendedFlag = ((indexEntryFlags & (1 << 14)) >> 14);
+                if (getLogger().isDebugEnabled())
+                    getLogger()
+                            .debug(
+                                    "Entry "
+                                            + entryIndex
+                                            + " has an extended flag of "
+                                            + indexEntryExtendedFlag);
+
+                // check that we parsed out the index entry extended flag correctly.
+                // this is more of an assertion than anything. It's already saved my bacon once.
+                if (indexEntryExtendedFlag != 0 && indexEntryExtendedFlag != 1) {
```

Comment:
LGTM claims this test is always false because of the second condition. However, given the comment I'm not sure what to do with it.

Reply:
I'm fine if removed, ideally this would have a test covering both cases. As general comment maybe we should use a library to parse it.

Reply:
You mean remove this block entirely?